data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -67,22 +72,26 @@
   private final List<DeleteFile> eqDeletes;
   private final Schema requiredSchema;
   private final Accessor<StructLike> posAccessor;
+  private final DeleteCounter counter;
 
   private PositionDeleteIndex deleteRowPositions = null;
-  private Predicate<T> eqDeleteRows = null;
 
-  protected DeleteFilter(String filePath, List<DeleteFile> deletes, Schema tableSchema, Schema requestedSchema) {
+  protected DeleteFilter(String filePath, List<DeleteFile> deletes, Schema tableSchema, Schema requestedSchema,
+      DeleteCounter counter) {
```

Comment:
Do we need to pass through the counter here? Seems like this should just be an internal member of delete filter that is created with the delete filter? Or is this just because we have no other way of getting the counter information back out of the applied delete filter?

Reply:
Yes, we need to pass the `DeleteCounter` in. The `DeleteCounter` is created in either `RowDataReader` or `BatchDataReader` and in their `open(FileScanTask)` method,  an instance of their `SparkDeleteFilter` static nested class is created for that `FileScanTask` with the `DeleteCounter`. We need a single counter in either `RowDataReader` or `BatchDataReader` which is used to aggregate the delete count over all the `FileScanTask`s.

Reply:
Update: The `DeleteCounter` is now created in `BaseReader` and passed to the instance of `BaseReader.SparkDeleteFilter` constructed in the `open(FileScanTask)` method `RowDataReader`/`BatchDataReader`. The argument still stands: we need a single counter in the reader which is used to aggregate the delete count over all the `FileScanTask`s.